src/Workspaces/Core/Portable/Workspace/Host/SourceFiles/DynamicFileInfo.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    /// <summary>
+    /// provides info on the given file
+    /// 
+    /// this will be used to provide dynamic content such as generated content from cshtml to workspace
+    /// we acquire this from <see cref="IDynamicFileInfoProvider"/> exposed from external components such as razor for cshtml
+    /// </summary>
+    internal sealed class DynamicFileInfo
```

Comment:
if we own this type, i'm not sure why other parts of the workspace needed maps from the filepaths in here to other filepaths.  This type shoud be able to contain all the information we need here.

Reply:
I'm referring to maps like: `private readonly Dictionary<string, string> _dynamicFilePathMaps = new Dictionary<string, string>();`

Instead of scattering all this data around, since we own *all* of it, we can just consolidate into unified representations that we then can operate on atomically.